test/test-utils.js

```diff
@@ -1,13 +1,125 @@
 'use strict';
 
+const jwt = require('../');
+const expect = require('chai').expect;
+const sinon = require('sinon');
+
+/**
+ * Correctly report errors that occur in an asynchronous callback
+ * @param {function(err): void} done The mocha callback
+ * @param {function(): void} testFunction The assertions function
+ */
+function asyncCheck(done, testFunction) {
+  try {
+    testFunction();
+    done();
```

Comment:
I guess we want this call (`done()`) outside of the try to avoid double called in case of error inside of it, right? (because we are passing the `done` from mocha)

Reply:
^ Silly thing, I'm also fine if you leave it as is, just let me know and I'll merge.

Reply:
Both options have different problems.

The way the code is currently means that if the `done()` call throws an error then `done(err)` will be called. This would be a double call to `done` which is generally considered a bad practice. In the case of Mocha, it will warn about the double call to `done`. Also, in this case anyways, for `done()` to fail, it would mean that Mocha is broken in some way.

By moving the `done()` call outside the `try` block then an error thrown by `done()` could be swallowed and not reported by Mocha or logged to the console.

I decided on what is there now, because I would rather Mocha tell the developer that something went wrong, then ignoring the error, or reporting that `done` was not called.

I think it's fine to keep things this way and try to fix the library later to avoid this problem altogether.